symcache/src/writer.rs

```diff
@@ -256,8 +256,11 @@ impl<W: Write> SymCacheWriter<W> {
         vmaddr: u64,
         symbol_iter: &mut Peekable<SymbolIterator>,
     ) -> Result<()> {
-        while let Some(&Ok(symbol)) = symbol_iter.peek() {
-            let sym_addr = symbol.addr() - vmaddr;
+        loop {
+            let sym_addr = match symbol_iter.peek() {
+                Some(&Ok(ref symbol)) => symbol.addr() - vmaddr,
+                _ => break,
```

Comment:
This change looks so strange to avoid two mutable borrows at the same time from `peek()` and `next()`.

Reply:
love borrowck